zhaquirks/inovelli/VZM31SN.py

```diff
@@ -1,578 +1,12 @@
-        ENDPOINTS: {
-            1: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: DeviceType.DIMMABLE_LIGHT,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    Identify.cluster_id,
-                    Groups.cluster_id,
-                    Scenes.cluster_id,
-                    OnOff.cluster_id,
-                    LevelControl.cluster_id,
-                    Metering.cluster_id,
-                    ElectricalMeasurement.cluster_id,
-                    Diagnostic.cluster_id,
-                    InovelliVZM31SNCluster,
-                    WWAH_CLUSTER_ID,
-                ],
-                OUTPUT_CLUSTERS: [
-                    Identify.cluster_id,
-                    OnOff.cluster_id,
-                    LevelControl.cluster_id,
-                    Ota.cluster_id,
-                    InovelliVZM31SNCluster,
-                ],
-            },
-            2: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: DeviceType.DIMMER_SWITCH,
-                INPUT_CLUSTERS: [Basic.cluster_id, Identify.cluster_id],
-                OUTPUT_CLUSTERS: [
-                    Identify.cluster_id,
-                    OnOff.cluster_id,
-                    LevelControl.cluster_id,
-                    InovelliVZM31SNCluster,
-                ],
-            },
-        },
-    }
-
-    device_automation_triggers = INOVELLI_AUTOMATION_TRIGGERS
-
-
-class InovelliVZM31SNv9(CustomDevice):
-    """VZM31-SN 2 in 1 Switch/Dimmer Module."""
-
-    signature = {
-        MODELS_INFO: [("Inovelli", "VZM31-SN")],
-        ENDPOINTS: {
-            1: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: DeviceType.DIMMER_SWITCH,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    Identify.cluster_id,
-                    Groups.cluster_id,
-                    Scenes.cluster_id,
-                    OnOff.cluster_id,
-                    LevelControl.cluster_id,
-                    Metering.cluster_id,
-                    ElectricalMeasurement.cluster_id,
-                    Diagnostic.cluster_id,
-                    INOVELLI_VZM31SN_CLUSTER_ID,
-                    WWAH_CLUSTER_ID,
-                ],
-                OUTPUT_CLUSTERS: [Identify.cluster_id, Ota.cluster_id],
-            },
-            2: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: DeviceType.DIMMER_SWITCH,
-                INPUT_CLUSTERS: [Identify.cluster_id],
-                OUTPUT_CLUSTERS: [
-                    Identify.cluster_id,
-                    OnOff.cluster_id,
-                    LevelControl.cluster_id,
-                    INOVELLI_VZM31SN_CLUSTER_ID,
-                ],
-            },
-        },
-    }
-
-    replacement = {
-        ENDPOINTS: {
-            1: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: DeviceType.DIMMABLE_LIGHT,
```

Comment:
Some of these devices also provide `DIMMER_SWITCH` as the device type. All VZM31-SN v1 quirks change the device type to `DIMMABLE_LIGHT`.
We should likely do that as well in the v2 quirk, although I don't think it should affect anything in ZHA.

Reply:
Ah, switch vs light entities is a big issue. Since we worked around in the past, we should still do that.

Reply:
Ugh, I thought just doing `replaces_endpoint(1, device_type=zha.DeviceType.DIMMABLE_LIGHT)` would be enough, but that removes all device reported clusters from that endpoint.

That should probably be addressed in zigpy. I doubt we ever want that.

Reply:
This should work with the next zigpy release that includes https://github.com/zigpy/zigpy/pull/1602. Calling the code in the above comment will only replace the endpoint type and profile (default is ZHA profile) then, not touch the existing device reported clusters anymore.

We'll need to address quirks compatibility with zigpy 0.80.0 first though: https://github.com/zigpy/zha-device-handlers/pull/4088
(and then bump the quirks repo again to a newer zigpy version including the linked PR).